### PR TITLE
Cleanup Dilation description.

### PR DESCRIPTION
**Pull request by meejah**
Convert to semantic newlines, some copy-editing, and add an introduction.

Describe Noise segmentation, message encoding as well.

### CODE REVIEW

**Comment by meejah** [comment]
CodeCov broken again: there's no code changes here!

File: docs/dilation-protocol.md [hunk 1]
```diff
@@ -203,20 +203,38 @@ Should each hint have its own priority, or just each endpoint?
...
 
-Everything beyond that point is a Noise protocol message, which consists of a 4-byte big-endian length field, followed by the indicated number of bytes.
-This uses the `NNpsk0` pattern with the Leader as the first party ("-> psk, e" in the Noise spec), and the Follower as the second ("<- e, ee").
-The pre-shared-key is the "dilation key", which is statically derived from the master PAKE key using HKDF.
+The encoding works like this: there is a 4-byte big-endian length field, followed by some number of Noise packets.
+There is no leading length field on each Noise packet: implementations MUST respect the Noise limits.
+So if the length field indicates a message bigger than 65536, the reader pulls 65536 bytes out of the stream, decrypts that blob as a Noise message, subtracts 65536 from the total and continues.
+The last Noise message will obviously be less than 65536 bytes.
+
+The entire decoded blob is then "one L2 message" and is develivered upstream.
```
Comment (vu3rdd) [inline docs/dilation-protocol.md:222]: small typo: `develivered` -> `delivered`?

File: docs/dilation-protocol.md [hunk 2]
```diff
@@ -237,17 +255,26 @@ All listening sockets may or may not be shut down (TODO: think about it).
...
-* the Noise session produces a valid decrypted frame (for Leader, this includes the Follower's KCM, and thus indicates a viable candidate for connection selection)
-* the Noise session reports a failed decryption
-* the TCP session is lost
+For developers attempting to understand the Python reference implementation (in `wormhole._dilattion` package):
+
+Internally, the overall endeavour is managed by the `Manager` object.
+For each generation, a single `Connection` object is created; this object manages the race between potential hints-based peer connections.
+A `DilatedConnctionProtocol` instance manages the Noise session itself.
+
+It knows via its `_role` attribute whether it is on the Leader or Follower side, which affects both the role is plays in the Noise pattern, and the reaction to receiving the handshake message / ephemeral key (for which only the Follower sends an empty KCM message).
```
Comment (vu3rdd) [inline docs/dilation-protocol.md:267]: "...both the role is plays ..." -> "...it plays..."?

File: docs/dilation-protocol.md [hunk 3]
```diff
@@ -237,17 +255,26 @@ All listening sockets may or may not be shut down (TODO: think about it).
...
+
+After that, the `DilatedConnectionProtocol` notifies the management obects in three situations:
+
+* the Noise session prodces a valid KCM message (`Connector` notified with `add_candidate()`).
```
Comment (vu3rdd) [inline docs/dilation-protocol.md:271]: produces?

**Review by vu3rdd — APPROVED** [review]
Left a few typos but overall, the text reads a lot clearer than before.

**Pull request merged**